src/Client/Updater.php

```diff
@@ -377,21 +382,26 @@ protected function getMetadataForTarget(string $target): ?TargetsMetadata
      * @return \GuzzleHttp\Promise\PromiseInterface<\Tuf\Metadata\TargetsMetadata>
      *   A promise wrapping the verified metadata for the role.
      */
-    private function fetchAndVerifyTargetsMetadata(string $role): PromiseInterface
+    private function fetchAndVerifyTargetsMetadata(string $role): TargetsMetadata
     {
+        if (isset($this->targetsMetadata[$role])) {
+            return $this->targetsMetadata[$role];
+        }
```

Comment:
I don't think we should change the return type here; let's just return an already-fulfilled promise. Or better yet, we can cache the promises; that way, even if a promise hasn't completed, we know we have a promise pending for that role.

Reply:
Switched to caching promises - this seems to be OK!